src/core/input_descriptor.rs

```diff
@@ -662,67 +660,57 @@ impl ConstraintsField {
-            .filter_map(|path| path.split(&['-', '.', ':', '@'][..]).last())
-            .map(|path| {
-                path.chars()
-                    .fold(String::new(), |mut acc, c| {
-                        // Convert camelCase to space-separated words with capitalized first letter.
-                        if c.is_uppercase() {
-                            acc.push(' ');
-                        }
-
-                        // Check if the field is snake_case and convert to
-                        // space-separated words with capitalized first letter.
-                        if c == '_' {
-                            acc.push(' ');
-                            return acc;
-                        }
-
-                        acc.push(c);
-                        acc
-                    })
-                    // Split the path based on empty spaces and uppercase the first letter of each word.
-                    .split(' ')
-                    .fold(String::new(), |desc, word| {
-                        let word =
-                            word.chars()
-                                .enumerate()
-                                .fold(String::new(), |mut acc, (i, c)| {
-                                    // Capitalize the first letter of the word.
-                                    if i == 0 {
-                                        if let Some(c) = c.to_uppercase().next() {
-                                            acc.push(c);
-                                            return acc;
-                                        }
-                                    }
-                                    acc.push(c);
-                                    acc
-                                });
-
-                        format!("{desc} {}", word.trim_end())
-                    })
-                    .trim_end()
-                    .to_string()
-            })
-            .collect()
+            .filter_map(|path| selector(path).ok())
+            .flatten()
+            .map(ToOwned::to_owned)
+            // NOTE: It is likely that only one of the paths will
+            // match the selector. Therefore, we're selecting only
+            // the first match that exists, if any.
+            //
+            // It may also be possible that multiple paths match,
+            // in which case there may be multiple raw fields
+            // that can be requested.
+            //
+            // As a result, it may be more acceptable to use `collect()` instead
+            // of `first()` to return all the possible requested fields.
+            .collect::<Vec<serde_json::Value>>()
+            .pop();
+
+        let purpose = self.purpose().map(ToOwned::to_owned);
+
+        let name = self
+            .name()
+            .map(ToOwned::to_owned)
+            // TODO: Add an "unknown field" if the name is not provided.
+            // Consider skipping or erroring on unknown fields.
+            .unwrap_or_default();
```

Comment:
No need to lose information. If the request doesn't include a name then propagate that Option into the RequestedField struct
```suggestion
        let name = self
            .name()
            .map(ToOwned::to_owned);
```